synapse/lib/storm.py

```diff
@@ -2862,6 +2862,95 @@ async def run(item):
             async for runitem in run(None):
                 yield runitem
 
+class BatchCmd(Cmd):
+    '''
+    Run a query with batched sets of nodes.
+
+    The batched query will have the set of inbound nodes available in the
+    variable $nodes.
+
+    This command also takes a conditional as an argument. If the conditional
+    is evaluates to true, the nodes returned by the batched query will be yielded,
+    if it evaluates to false, the inbound nodes will be yielded after executing the
+    batched query.
+
+    NOTE: This command is intended to facilitate use cases such as queries to external
+          APIs with aggregate node values to reduce quota consumption. As this command
+          interrupts the node stream, it should be used carefully to avoid unintended
+          slowdowns in the pipeline.
+
+    Example:
+
+        // Execute a query with batches of 5 nodes, then yield the inbound nodes
+        batch $lib.false --size 5 { $lib.print($nodes) }
+    '''
+    name = 'batch'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('cond', help='The conditional value for the yield option.')
```

Comment:
I think this bears close enough resemblance to use `--yield` and make it optional

Reply:
Changing it to just be a `--yield` switch feels a bit clunkier to me for the intended purpose of using this in commands, it seems like most commands would end up always using `batch --yield` with a :
```
if (not $cmdopts.yield) { spin }
fini { if $cmdopts.yield { yield $nodes } }
```
at the end of the batch query to get the correct behavior

Reply:
leaving as cond